pkg/provider/prefix/provider.go

```diff
@@ -288,6 +289,10 @@ func (p *ipv4PrefixProvider) CreateIPv4PrefixAndUpdatePool(job *worker.WarmPoolJ
 	if err != nil {
 		p.log.Error(err, "failed to create all/some of the IPv4 prefixes", "created resources", resources)
 		didSucceed = false
+		if strings.HasPrefix(err.Error(), utils.InsufficientCidrBlocksReason) {
```

Comment:
This adds a dependency on EC2 API error. It is ok for now since it is just for sending a node event. Can you add a comment here to make a note.

Reply:
Sounds good, comment added.